homeassistant/components/suez_water/coordinator.py

```diff
@@ -72,19 +90,157 @@ def map_dict(param: dict[date, float]) -> dict[str, float]:
-                    this_year_overall=aggregated.current_year,
-                    history=map_dict(aggregated.history),
-                ),
-                price=(await self._suez_client.get_price()).price,
-            )
         except PySuezError as err:
-            raise UpdateFailed(f"Suez data update failed: {err}") from err
+            raise UpdateFailed("Suez coordinator error communicating with API") from err
+
+        price = None
+        try:
+            price_result = await self._suez_client.get_price()
+            if price_result:
+                price = price_result.price
+        except PySuezError:
+            _LOGGER.debug("Failed to fetch water price", stack_info=True)
+
+        try:
+            await self._update_statistics(price)
+        except PySuezError as err:
+            raise UpdateFailed("Failed to update suez water statistics") from err
+
         _LOGGER.debug("Successfully fetched suez data")
-        return data
+        return SuezWaterData(
+            aggregated_value=aggregated.value,
+            aggregated_attr=SuezWaterAggregatedAttributes(
+                this_month_consumption=map_dict(aggregated.current_month),
+                previous_month_consumption=map_dict(aggregated.previous_month),
+                highest_monthly_consumption=aggregated.highest_monthly_consumption,
+                last_year_overall=aggregated.previous_year,
+                this_year_overall=aggregated.current_year,
+                history=map_dict(aggregated.history),
+            ),
+            price=price,
+        )
+
+    async def _update_statistics(self, current_price: float | None) -> None:
+        """Update daily statistics."""
+        _LOGGER.debug("Updating statistics for %s", self._water_statistic_id)
+
+        water_last_stat = await self._get_last_stat(self._water_statistic_id)
+        cost_last_stat = await self._get_last_stat(self._cost_statistic_id)
+
+        consumption_sum = 0.0
+        cost_sum = 0.0
+        last_stats = None
+
+        if water_last_stat is not None:
+            last_stats = datetime.fromtimestamp(water_last_stat["start"]).date()
+            if water_last_stat["sum"] is not None:
+                consumption_sum = water_last_stat["sum"]
+        if cost_last_stat is not None:
+            if cost_last_stat["sum"] is not None and cost_last_stat["sum"] is not None:
+                cost_sum = cost_last_stat["sum"]
+        if last_stats is None:
+            last_stats = datetime.now().date()
+            last_stats = last_stats.replace(year=last_stats.year - 2)
+
+        _LOGGER.debug(
+            "Updating suez stat since %s for %s",
+            str(last_stats),
+            water_last_stat,
+        )
+        usage = await self._suez_client.fetch_all_daily_data(
+            since=last_stats,
+        )
+        if usage is None or len(usage) <= 0:
+            _LOGGER.debug("No recent usage data. Skipping update")
+            return
```

Comment:
both, None and an empty list will assert to False
```suggestion
        if not (
            usage := await self._suez_client.fetch_all_daily_data(
                since=last_stats,
            )
        ):
            _LOGGER.debug("No recent usage data. Skipping update")
            return
```

Reply:
I'm not used to having assingnation in if's :sweat_smile: